.openpublishing.redirection.json

```
@@ -1178,6 +1178,11 @@
             "source_path": "aspnetcore/blazor/forms-validation.md",
             "redirect_url": "/aspnet/core/blazor/forms-and-input-components",
             "redirect_document_id": false
+        },
+        {
+            "source_path": "aspnetcore/core/grpc/json-transcoding.md",
+            "redirect_url": "/aspnet/core/grpc/httpapi",
+            "redirect_document_id": false
```

source_path is the path in the repo to the Markdown file of the article that has the URL you want to redirect from.
redirect_url is the redirect-to-URL

Try this:

```suggestion
            "source_path": "aspnetcore/grpc/httpapi.md",
            "redirect_url": "/aspnet/core/grpc/json-transcoding",
            "redirect_document_id": false
```

---

> source_path is the path in the repo to the Markdown file of the article that has the URL you want to redirect from.
redirect_url is the redirect-to-URL

Does that mean I shouldn't rename the original httpapi.md file? I renamed it to json-transcoding.md.

---

Redirects work with the build URLs so you can test your redirect before merging.

---

>Does that mean I shouldn't rename the original httpapi.md file? I renamed it to json-transcoding.md.

No, the rename and redirect should work.